src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx

```diff
@@ -61,53 +61,51 @@ export const savePollInterval = (refreshIntervalCookie: string, pollInterval: nu
 const initializeLocalStorage = (props: AcmAutoRefreshSelectProps) => {
     const initialValue = props.pollInterval
     const key = props.refreshIntervalCookie ?? DEFAULTS.refreshIntervalCookie
+    const defaultValue = (props.initPollInterval ?? DEFAULTS.initPollInterval) * 1000
 
     return useState<number>((): number => {
-        if (initialValue) {
-            window.localStorage.setItem(key, `${initialValue}`)
+        if (initialValue != null /* initialValue can be 0 for refresh disabled */) {
+            savePollInterval(key, initialValue)
             return initialValue
         } else if (window && window.localStorage && window.localStorage.getItem(key)) {
             /* istanbul ignore next */
-            const value =
-                window.localStorage.getItem(key) ?? `${(props.initPollInterval ?? DEFAULTS.initPollInterval) * 1000}`
+            const value = window.localStorage.getItem(key) ?? `${defaultValue}`
             return parseInt(value, 10)
         }
-        window.localStorage.setItem(key, `${(props.initPollInterval ?? DEFAULTS.initPollInterval) * 1000}`)
-        return (props.initPollInterval ?? DEFAULTS.initPollInterval) * 1000
+        savePollInterval(key, defaultValue)
+        return defaultValue
     })
 }
 
 export function AcmAutoRefreshSelect(props: AcmAutoRefreshSelectProps) {
     const [isOpen, setOpen] = useState<boolean>(false)
     const [selected, setStoredValue] = initializeLocalStorage(props)
-    const [addedListener, setAddedListener] = useState<boolean>(false)
-    const [docHidden, setDocHidden] = useState<boolean>(false)
+    const [initialFetchCalled, setInitialFetchCalled] = useState<boolean>(false)
+    const [docHidden, setDocHidden] = useState<boolean>(window.document.hidden)
     const onVisibilityChange = () => {
         setDocHidden(window.document.hidden)
     }
-    if (!addedListener) {
-        document.addEventListener('visibilitychange', onVisibilityChange)
-        setAddedListener(true)
-    }
+
```

Comment:
Should be done inside an effect. Moved event listener addition/removal to its own effect and removed `addedListener` state to simplify logic.